bundles/org.openhab.binding.bluetooth.hdpowerview/src/main/java/org/openhab/binding/bluetooth/hdpowerview/internal/shade/ShadeHandler.java

```diff
@@ -0,0 +1,533 @@
+                    case CHANNEL_SHADE_PRIMARY:
+                        command = dataReader.getPrimary();
+                        break;
+                    case CHANNEL_SHADE_SECONDARY:
+                        command = dataReader.getSecondary();
+                        break;
+                    case CHANNEL_SHADE_TILT:
+                        command = dataReader.getTilt();
+                        break;
+                }
+            }
+        }
+
+        // convert up/down commands to position command
+        if (command instanceof UpDownType updown) {
+            command = UpDownType.DOWN == updown ? PercentType.ZERO : PercentType.HUNDRED;
+        }
+
+        if (command instanceof PercentType percent) {
+            Capabilities capabilities = this.capabilities;
+            if (capabilities == null) {
+                return;
+            }
+
+            try {
+                switch (channelUID.getId()) {
+                    case CHANNEL_SHADE_PRIMARY:
+                        if (capabilities.supportsPrimary()) {
+                            scheduleWritePosition(new ShadeDataWriter().withSequence(writeSequence++)
+                                    .withPrimary(percent.doubleValue()).getEncrypted(getEncryptionKey()));
+                        }
+                        break;
+
+                    case CHANNEL_SHADE_SECONDARY:
+                        if (capabilities.supportsSecondary()) {
+                            scheduleWritePosition(new ShadeDataWriter().withSequence(writeSequence++)
+                                    .withSecondary(percent.doubleValue()).getEncrypted(getEncryptionKey()));
+                        }
+                        break;
+
+                    case CHANNEL_SHADE_TILT:
+                        if (capabilities.supportsTiltOnClosed() || capabilities.supportsTilt180()
+                                || capabilities.supportsTiltAnywhere()) {
+                            scheduleWritePosition(new ShadeDataWriter().withSequence(writeSequence++)
+                                    .withTilt(percent.doubleValue()).getEncrypted(getEncryptionKey()));
+                        }
+                        break;
+                }
+            } catch (InvalidKeyException | IllegalArgumentException | NoSuchAlgorithmException | NoSuchPaddingException
+                    | InvalidAlgorithmParameterException | IllegalBlockSizeException | BadPaddingException e) {
+                logger.warn("handleCommand() device={} error={}", device.getAddress(), e.getMessage(),
+                        logger.isDebugEnabled() ? e : null);
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        super.initialize();
+        configuration = getConfigAs(ShadeConfiguration.class);
+        updateProperty(PROPERTY_HOME_ID, Integer.toHexString(homeId).toUpperCase());
+        activityTimeout = Instant.now().plusSeconds(configuration.pollingDelay * 2);
+
```

Comment:
I would suggest to validat the configuration and set the thing state to `CONFIGURATION_ERROR` accordingly.

Reply:
See above.

Reply:
This is a little different I guess. What if the user enters the wrong or malformed address parameter?

Reply:
- Apropos 'malformed' Ok, I will add a regex check for MAC addresses.
- Apropos 'wrong' the thing goes offline (communication error) since the bridge won't see the BLE beacon.